server/src/controllers/susbcriptionController.ts

```diff
@@ -128,6 +132,177 @@ subscriptionController.get('/season/:id', async (req, res: Response) => {
+  try {
+    const today = new Date();
+    const formattedDate = today.getFullYear()*(today.getMonth()<9?10000:100000)+(today.getMonth()+1)*100+today.getDate();
+    const seasons = await prisma.seasons.findMany({
+      where: {
+        deletedat: null,
+        seasonsubscriptiontypes: {
+          some: {
+            deletedat: null,
+          },
+        },
+        enddate: {gte: formattedDate},
+        events: {
+          some: {
+            subscriptioneligible: true,
+            deletedat: null,
+          },
+        },
+      },
+    });
+    return res.json(seasons);
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      return res.status(400).json({error: error.message});
+    }
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      return res.status(400).json({error: error.message});
+    }
+    return res.status(500).json({error: 'Internal Server Error'});
+  }
+});
+
+/**
+ * @swagger
+ * /2/subscription-types/active-subscriptions/{seasonid}:
+ *   get:
+ *     summary: get season including all events and available subscriptions
+ *     tags:
+ *     - Subscription API
+ *     parameters:
+ *     - $ref: '#/components/parameters/seasonid'
+ *     responses:
+ *       200:
+ *         description: season successfully fetched
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               $ref: '#/components/schemas/Season'
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   description: Error message from the server.
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+subscriptionController.get('/active-subscriptions/:seasonid', async (req: Request, res: Response) => {
+  try {
+    const seasonid = +req.params.seasonid;
+
+    const season = await prisma.seasons.findUnique({
+      where: {
+        seasonid: seasonid,
+        deletedat: null,
+      },
+      include: {
+        seasonsubscriptiontypes: {
+          where: {
+            deletedat: null,
+          },
+          include: {
+            subscriptiontype: true,
+            subscriptions: {
+              where: {
+                refund: null,
+              },
+            },
+          },
+        },
+        events: {
+          where: {
+            deletedat: null,
+          },
+          include: {
+            eventinstances: {
+              where: {
+                deletedat: null,
+              },
+              orderBy: {
+                eventdate: 'asc',
+              },
+            },
+          },
+        },
+      },
+    });
+
+    if (!season) {
+      return res.status(400).json({error: 'Season does not exist'});
```

Comment:
Since this is a get request, does it make sense for this to be a 404 if the season isn't found?

Reply:
Updated